pivotreport.php

```diff
@@ -188,3 +188,29 @@ function pivotreport_civicrm_entityTypes(&$entityTypes) {
     'table' => 'civicrm_pivotreport_config',
   ];
 }
+
+/**
+ * Implements hook_civicrm_alterAPIPermissions()
+ */
+function pivotreport_civicrm_alterAPIPermissions($entity, $action, &$params, &$permissions) {
+  switch (true) {
+    case $entity == 'setting' && $action == 'get':
+      $return = $params['return'];
+
+      if (in_array('weekBegins', $return) || in_array('fiscalYearStart', $return)) {
+        $permissions[$entity][$action] = array('access CiviCRM pivot table reports');
+      }
+      break;
+
+    case $entity == 'pivot_report':
+    case $entity == 'pivot_report_config':
+      $permissions[$entity][$action] = array('access CiviCRM pivot table reports');
```

Comment:
why to use the hook to alter the extension itself APIs, you can just alter the extension code.

Reply:
I'm not sure how I'd do this, as I understand, the check is done outside the extension, and all ajax API calls require 'Administer CiviCRM' permission, unless changed with this hook. Official CiviCRM documentation specifies this should be the way to change API call permissions on your own extension. If there's another way to do it, could you point out how?

Reply:
ah I see, I didn't work on Civi REST API before so  I didn't knew that.